tools/payments/cmd/disaster/main.go

```diff
@@ -0,0 +1,133 @@
+/*
+disaster is a disaster recovery command that will download the encrypted secrets file from S3 and decrypt it using locally-possessed shares.
+
+Usage:
+
+disaster [flags] [shareFiles...]
+
+The flags are:
+
+	-k
+		Location on file system of the operators private ED25519 signing key in PEM format.
+	-e
+		The environment to which the operator is sending approval for transactions.
+	-b
+		The s3 from which to retrieve the configuration
+	-f
+		The filename to get from the s3 bucket
+*/
+package main
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"filippo.io/age"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/brave-intl/bat-go/libs/requestutils"
+	"github.com/hashicorp/vault/shamir"
+)
+
+func main() {
+	ctx := context.Background()
+
+	// command line flags
+	operatorKey := flag.String(
+		"k", "test/private.pem",
+		"the operator's key file location (ed25519 private key) in PEM format")
+
+	s3Bucket := flag.String(
+		"b", "",
+		"the s3 bucket to upload to")
+
+	s3Object := flag.String(
+		"o", "",
+		"the s3 object name for output")
+
+	env := flag.String(
+		"e", "local",
+		"the environment to which the tool will interact")
+
+	verbose := flag.Bool(
+		"v", false,
+		"view verbose logging")
+
+	flag.Parse()
+
+	args := flag.Args()
+	if len(args) < 2 {
+		log.Fatal("insufficient shares provided")
+	}
+
+	disaster(ctx, *operatorKey, *s3Bucket, *s3Object, args, *env, *verbose)
+}
+
+func disaster(ctx context.Context, key, bucket, file string, shareFilenames []string, env string, verbose bool) {
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		log.Fatalf("failed to load default aws config: %v", err)
+	}
+	s3Client := s3.NewFromConfig(cfg)
+	output, err := s3Client.GetObject(ctx, &s3.GetObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(file),
+	})
+	if err != nil {
+		log.Fatalf("failed to upload configuration: %v", err)
+	}
+
+	bodyBytes, err := requestutils.Read(ctx, output.Body)
+	if err != nil {
+		log.Fatalf("failed to read response body: %w", err)
+	}
+	var shares [][]byte
+	for _, f := range shareFilenames {
+		dat, err := os.ReadFile(f)
+		if err != nil {
+			log.Fatalf("failed to read share file: %w", err)
+		}
+		shares = append(shares, dat)
+	}
+	recoveredData, err := decryptRecoveryData(bodyBytes, shares)
```

Comment:
is the idea that the share files would have been decrypted with `age` in advance?

it may make sense to have a 2 step process:
1. each operator other than the recovery lead decrypts their share and re-encrypts to the recovery lead's key
2. the recovery lead runs this

Reply:
This seems like the easiest, but less transparent approach here. 

The alternative would require server side changes to coordinate this where multiple operators submit authorization and once enough have submitted then the SSS output can be encrypted to the operators key and downloaded. The only advantage I see to that approach is that there's a semi-record on hand of the DR process occurring. 

Maybe a hybrid solution here is to have a record when the encrypted config secrets are pulled from the s3 bucket as an easy alternative?

Reply:
We need DR to work in a situation where even the enclave is unavailable/down/broken, so coordination across it isn't viable. I think this approach of re-encrypting with a leader key and sharing that key out of band is a good solution. We could make it transmit the leader-encrypted share over s3. But again, in a DR scenario it's probably better to assume as little as possible. In the current implementation the only dependency is fetching the secrets from s3 itself. Nothing else.

Reply:
Makes sense, I hadn't thought through that as well, but agree with the requirement now that you call it out.